src/Layouts/Partials/FillItem.vala

```diff
@@ -159,24 +217,37 @@ public class Akira.Layouts.Partials.FillItem : Gtk.Grid {
         opacity_container = new Akira.Partials.InputField (
             Akira.Partials.InputField.Unit.PERCENTAGE, 7, true, true);
         opacity_container.entry.sensitive = true;
-        opacity_container.entry.text = (alpha / 255 * 100).to_string ();
+        opacity_container.entry.text = Math. round ((double) alpha / 255 * 100).to_string ();
         opacity_container.entry.bind_property (
             "text", model, "alpha",
             BindingFlags.BIDIRECTIONAL | BindingFlags.SYNC_CREATE,
-            (binding, srcval, ref targetval) => {
-                double src = double.parse (srcval.dup_string ());
+            (binding, entry_text_val, ref model_alpha_val) => {
+                double src = double.parse (entry_text_val.dup_string ());
+
                 if (src > 100) {
+                    src = 100.0;
                     opacity_container.entry.text = "100";
-                    return false;
                 } else if (src < 0) {
+                    src = 0.0;
                     opacity_container.entry.text = "0";
-                    return false;
                 }
-                targetval.set_double (src / 100);
+
+                color_set_manually = true;
+
+                int alpha_int_value = (int) (src / 100 * 255);
+
+                model_alpha_val.set_int (alpha_int_value);
+
+                set_button_color (null, alpha_int_value);
+
                 return true;
-            }, (binding, srcval, ref targetval) => {
-                double src = (double) srcval / 255;
-                targetval.set_string (("%0.1f").printf (src * 100));
+            }, (binding, model_alpha_val, ref entry_text_val) => {
+                double src = (double) model_alpha_val.get_int () / 255;
+
+                src = Math.round (src * 100);
+
+                entry_text_val.set_string (("%f").printf (src * 100));
+
```

Comment:
Why the `src` is multiplied by `100` twice?

Reply:
Typo, removed!